custom_components/ha_washdata/ml/engine.py

```
@@ -0,0 +1,125 @@
+Each model consumes a feature mapping whose keys are the model's
+``FEATURE_COLUMNS``; the integration computes those from live data per the
+``*_feature_contract.json`` files shipped alongside the model modules.
+"""
+
+from __future__ import annotations
+
+import importlib
+import json
+import logging
+from pathlib import Path
+from typing import Mapping
```

Ruff (0.15.20)

[warning] Import from `collections.abc` instead: `Mapping`

Import from `collections.abc`

(UP035)